pandas/tests/test_strings.py

```diff
@@ -469,12 +436,11 @@ def test_cat_on_filtered_index(self):
 
         str_year = df.year.astype('str')
         str_month = df.month.astype('str')
-        str_both = str_year.str.cat(str_month, sep=' ', join='left')
```

Comment:
why are you removing the join here? is it not needed? should these be parameterized by join type?

Reply:
In the course of #20347, the warnings were originally always issues whenever `join` was not specified - and this test correspondingly adapted. However, the warnings were later reduced to only be emitted in case of different indexes between caller and others, so in this case, it's not necessary.